_posts/2023-03-12-gs_00008_programming_II.md

```diff
@@ -0,0 +1,106 @@
+---
+
+Hello programming newbies! Welcome back to the second part of our programming
+tutorial. Last time we explained why we need programming languages. This time we
+want to take an example from our daily life to discuss a few general programming
+concepts. Imagine you wake up Sunday morning and want to make some pancakes. A
+pretty simple recipe could be: "Mix 200g flour, 2 eggs, and 200ml milk. Then,
+bake it in a pan". Let's see what this has to do with programming.
+
+## Pancakes for a programmer
+
+First, it should be clear that you have to follow the recipe **sequentially**.
+You cannot put anything into the pan before you have mixed the ingredients. The
+same holds for programs. They run sequentially. There are also programs running
+in parallel. Imagine a friend helps you with the pancake. You can measure 200ml
+milk and 200g flour in parallel, but as a whole, you still need to follow the
+sequential order (the pan still needs to wait). Parallel programs are similar,
+but here we want to assume pure sequential programs.
+
+Let's start with the eggs. We have to take an egg, break it, and put it into a
+bowl. Also, we need to throw the eggshell into the trash. For the second egg, we
+have to do exactly the same. In a programming language, you can define a
+**function** _put_egg_in_bowl_ to define a sequence that is used multiple times.
+Next, you can **call** this function in a **loop** for every egg. So, a program
+could look like this:
+
+    function put_egg_in_bowl:
+        take egg
+        break egg
+        put it into bowl
+        throw shell into trash
+
+    for 2 times:
+        put_egg_in_bowl
+
+Note, that we use indentation (here we use 4 spaces) to show which instructions
+belong to our function and loop. After the eggs, we want to weigh the flour. How
+do you usually do it? You watch the scale. Until your target weight of 200g is
+not reached you put more flour into the bowl. In programming languages, there
+exist two types of loops. For the one we used above, we know how often we need
+to repeat it. This is called a **for-loop**. For weighing the flour, this is not
+very useful since you don't know how often you have to put more. We need a loop
+that continues while a condition is true. This is called **while-loop**:
+
+    while flour < 200g:
+        put more flour
+
+We can continue very similarly for the milk:
+
+    while milk < 200ml:
+        put more milk
+
+_flour_ and _milk_ represent the current amount. In programming languages, we
+call these **variables**. In simple words, variables can store any kind of
+information.
+
+<div class='note-exercise' markdown="1">
+Exercise:
+
+Can you continue with the rest of the pancake?
+
+</div>
+
+There is one last question. Did you like the pancake? If yes you might want to
+recommend it to a friend. To implement such a question you can use an
+**if-else-statement**:
+
+    if pancake is delicious:
+        recommend the recipe
+    else:
+        delete the recipe
```

Comment:
The _if_ statement checks whether a condition is true. In our example, it is checked whether the pancake is delicious. If that condition is true, the instructions indented below the if statement will be executed, in our case recommending the recipe. If the pancakes didn't taste good, the instructions below the _else_ statement will be executed, here, deleting the recipe.